[GStreamer][Release] registerTrackId(): alternative IDs discarded

ntrrgc opened the pull request:
#### da44fa7432e25045dc0183848b1d0c3837530011
<pre>
[GStreamer][Release] registerTrackId(): alternative IDs discarded
<a href="https://bugs.webkit.org/show_bug.cgi?id=289245">https://bugs.webkit.org/show_bug.cgi?id=289245</a>

Reviewed by Philippe Normand.

Move side effects away from the assert so that they don&apos;t get removed in
release builds.

* Source/WebCore/platform/graphics/gstreamer/mse/MediaSourcePrivateGStreamer.cpp:
(WebCore::MediaSourcePrivateGStreamer::registerTrackId):

Canonical link: <a href="https://commits.webkit.org/291711@main">https://commits.webkit.org/291711@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/92dca2293c64be79d5686200da80390c1ef5fcef

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/93741 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/13315 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/3052 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/98747 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/44267 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/95791 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/13610 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/21757 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/71563 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/28935 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/96743 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/10131 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/84728 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/51897 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/9812 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/2356 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/43582 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/80085 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/2424 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/100782 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/20793 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/15172 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/80580 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/21045 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/80671 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/79918 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/19894 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/24466 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/1827 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/13950 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/20777 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/25955 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/20464 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/23924 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/22205 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

ntrrgc commented:
```
/app/webkit/Source/WebCore/platform/graphics/gstreamer/mse/MediaSourcePrivateGStreamer.cpp: In member function ‘WebCore::TrackID WebCore::MediaSourcePrivateGStreamer::registerTrackId(WebCore::TrackID)’:
/app/webkit/Source/WebCore/platform/graphics/gstreamer/mse/MediaSourcePrivateGStreamer.cpp:235:10: error: variable ‘result’ set but not used [-Werror=unused-but-set-variable]
  235 |     auto result = m_trackIdRegistry.add(assignedId);
      |          ^~~~~~
```

Right, have to do the maybe_unused thing...

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/92dca2293c64be79d5686200da80390c1ef5fcef. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/42000

ntrrgc committed:
[GStreamer][Release] registerTrackId(): alternative IDs discarded
https://bugs.webkit.org/show_bug.cgi?id=289245

Reviewed by Philippe Normand.

Move side effects away from the assert so that they don't get removed in
release builds.

* Source/WebCore/platform/graphics/gstreamer/mse/MediaSourcePrivateGStreamer.cpp:
(WebCore::MediaSourcePrivateGStreamer::registerTrackId):

Canonical link: https://commits.webkit.org/291711@main

aperezdc commented:
Backported into `webkitglib/2.48` as commit 257f420e1ca60c4e5b4f3dde71f7e7c99f4d74ce

aperezdc commented:
Backported into `webkitglib/2.46` as commit 86bdb2b0d7037d178b8a5fd4cb8748accd12d240
